packages/core/src/agent/message-list/index.ts

```diff
@@ -1068,16 +1060,22 @@ export class MessageList {
       if (latestMessage.createdAt.getTime() < messageV2.createdAt.getTime()) {
         latestMessage.createdAt = messageV2.createdAt;
       }
-      if (!latestMessage.content.content && messageV2.content.content) {
-        latestMessage.content.content = messageV2.content.content;
-      }
-      if (
-        latestMessage.content.content &&
-        messageV2.content.content &&
-        latestMessage.content.content !== messageV2.content.content
-      ) {
-        // Match what AI SDK does - content string is always the latest text part.
-        latestMessage.content.content = messageV2.content.content;
+      // Extract text content from parts for comparison and update
+      const latestTextContent = latestMessage.content.parts.find(p => p.type === 'text')?.text;
+      const messageV2TextContent = messageV2.content.parts.find(p => p.type === 'text')?.text;
```

Comment:
looks like this behaviour doesn't match what it did before, it said content.content was the latest text part, but now we're grabbing the first one

I wonder though, do we even need any of the code here anymore?